### DISCUSSION TITLE
Is there any way to decrypt information from database and then show this information in form that is opens after frmWelcome?

### DISCUSSION THREAD

**Discussion by seellyouat** [idpNET/secure-facial-recognition-security-system-using-emguCV#2] [Ideas]
Is there a way to create a form after succsesfull login to show all the registered users in this form?Or how even do decrypting a data from database?

**Comment by idpNET** [comment]
> Is there a way to create a form after succsesfull login to show all the registered users in this form?Or how even do decrypting a data from database?

Greetings,
Absolutely!
Users' info is saved into the database as follows:

    Id (auto-increment primary key)
    Username (plain text)
    UnderPasscode (as hashed byte[] which can not be decrypted, but verifiable)
    UserImage (encrypted byte[] which can be decrypted and loaded as an image; take a look at the LoadTrainingData method from FaceDetectionOperations class)
    UserID (as byte[] used in passcode verification )

Using the information above, you can easily iterate through database rows to make a list of registered users and use it as the way you wish. I suggest you taking a look at the variables in DatabaseOperations class. You can define a list of string instead of the UserNameValue string variable to hold all users' name in a list and iterate through it

Good Luck! Please let me know if you need help.

**Reply by seellyouat** [reply]
Thank you very much for your answer! I'm new to c# and github and I'm a student, just got very interested in your project, I find it a very good 2fa authentication method by using password and face identification, it's actually helped me to better understand usage of EmguCV and how computer vision is working for identification.

**Reply by idpNET** [reply]
> Thank you very much for your answer! I'm new to c# and github and I'm a student, just got very interested in your project, I find it a very good 2fa authentication method by using password and face identification, it's actually helped me to better understand usage of EmguCV and how computer vision is working for identification.

Thanks a lot! I wish you success
